implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicMeterRegistry.java

```diff
@@ -315,26 +412,78 @@ public Builder httpClient(HttpSender httpClient) {
             return this;
         }
 
+        public Builder insights(Insights insights) {
+            this.insights = insights;
+            return this;
+        }
+
         public NewRelicMeterRegistry build() {
-            return new NewRelicMeterRegistry(config, clock, threadFactory, httpClient);
+            return new NewRelicMeterRegistry(config, clock, threadFactory, httpClient, insights);
         }
     }
 
-    private class Attribute {
-        private final String name;
-        private final Object value;
+    /**
+     * POJO representation of an Insights event.
+     */
+    // VisibleForTesting
+    static class NewRelicEvent {
```

Comment:
To allow the APM/API integrations to share the majority of the code, a DTO to represent the event was created